src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -253,7 +253,11 @@ private protected sealed override TModel TrainModelCore(TrainContext context)
         {
             Host.CheckValue(context, nameof(context));
             var initPredictor = context.InitialPredictor;
-            var initLinearPred = initPredictor as LinearModelParameters ?? (initPredictor as CalibratedPredictorBase)?.SubPredictor as LinearModelParameters;
+            LinearModelParameters initLinearPred = null;
+            if (initPredictor is LinearModelParameters)
+                initLinearPred = (LinearModelParameters)initPredictor;
```

Comment:
Can be simplfiied to

```csharp
if (initPredictor is LinearModelParameters linModel)
    initLinearPred = linModel;
```

Same with code below.

Reply:
Sure. Many thanks for your education.

---
In reply to: [254540256](https://github.com/dotnet/machinelearning/pull/2435#discussion_r254540256) [](ancestors = 254540256)